src/docs/arch-column-row.dox

```diff
@@ -0,0 +1,93 @@
+any sequence of the same value with a count and value indicator. However, this makes the access to a
+specific record more complicated, it makes impossible to jump to a record in a leaf page. Finally,
+when reading and writing column-stores, only the columns required for any particular query are
+maintained in memory.
+
+@section Fixed length column-stores
+
+Fixed length column stores are very different internally and usually serve specific purposes as it
+is a <a href="https://en.wikipedia.org/wiki/Bitmap"><b>Bitmap</b></a>. It makes fixed length
+column-stores efficient to retrieve a value at a given offset, the use of
+<a href="https://en.wikipedia.org/wiki/Bloom_filter"><b>Bloom filter</b></a> is probably one of the
+best examples.
+
+@section Performance
+
+Generally, row-stores are faster for queries where all of the columns are required by every lookup
+because there is only a single set of internal pages to read into the cache and search.
+Column-stores are faster when most queries require only a subset of the columns because columns can
+be separated into multiple files and only the columns being returned need to be present in the
+cache. In fact, it is possible to split a table into multiple files using a combination of
+row-stores and column-stores, so this benefit can be gained regardless of the access method choice,
+see @ref schema_column_groups for more information. In both cases, for better efficiency, one should
+define column groups according to queries that will be mostly performed. Finally, row-stores and
+column-stores can be mixed-and-matched at the table level: for example, a row-store index can be
+created on a column-store table. It is worth noting that MongoDB does not use the rich schema being
+described. It stores a blob of data in a row-store and manages encoding and decoding itself.
+
+@section Internal usage
+
+Internally, row-stores and column-stores use a common \c WT_BTREE structure. The fundamental
+difference is that the \c WT_BTREE->type field is set \c BTREE_ROW for row-stores and \c BTREE_COL
+for column-stores. Internal functions that navigate, access and manipulate B-Trees have code
+sprinkled throughout that is conditional on \c WT_BTREE->type.
+
+<table>
+@hrow{, Row Store, Column Store - Variable Length, Column Store - Fixed Length}
+@row{Characteristics,,,,}
+@row{Internal representation, B-Tree (\c WT_BTREE), B-Tree (\c WT_BTREE), B-Tree (\c WT_BTREE)}
+@row{B-Tree type, Generalized (\c BTREE_ROW), Specialized (\c BTREE_COL_VAR), Specialized (\c BTREE_COL_FIX)}
+@row{Leaf pages, Key count equal to half the number of physical entries (unless all empty values flag is set where key count is equal to the number of physical entries), Only the first key is stored, Only the first key is stored}
+@row{Record key,Anything (byte-string), prefix compressed or overflow objects, Record id (64 bit unsigned integer), Record id (64 bit unsigned integer)}
```

Comment:
The commas don't line up for me. I'm expecting to see 'Record id' as the entries for the two column store.
`prefix compressed or overflow objects` looks like the entry for the varlength col store.

BTW, it's helpful for reviewers to attach a png or html of any chart/pic so they don't have to download the branch/build to give a review.

Reply:
Thanks for spotting the extra comma.
Yes, I will do that, sorry, it is not really nice to read the table in this format.

Reply:
See a716ab407a3f494009105e9d8b8dbcf329f5e226